themes/bootstrap3/templates/search/facet-list.phtml

```diff
@@ -28,36 +26,45 @@
     <label><?=$this->translate('Sort') ?></label>
     <div class="btn-group">
       <?php foreach ($this->sortOptions as $key => $sort): ?>
-        <a href="<?=$urlBase . '&amp;facetpage=1&amp;facetsort=' . urlencode($key) ?>" class="btn btn-default js-facet-sort<?php if ($this->sort == $key): ?> active<?php endif; ?>" data-sort="<?=$key ?>" data-lightbox-ignore><?=$this->translate($sort) ?></a>
+        <a href="<?=$this->escapeHtmlAttr($urlBase . '&facetpage=1&facetsort=' . urlencode($key) . '&contains=' . urlencode($contains)) ?>" class="btn btn-default js-facet-sort<?php if($this->sort == $key): ?> active<?php endif; ?>" data-sort="<?=$key ?>"><?=$this->translate($sort) ?></a>
       <?php endforeach; ?>
     </div>
+
+    <?php if (method_exists($this->results->getParams(), 'setFacetContains')): ?>
+      <label><?=$this->transEsc('Filter') ?></label>
```

Comment:
I'm not an accessibility expert, so I'd appreciate input from @pasitiis, @crhallberg or anyone else interested -- but I wonder if this would be slightly more accessible if we associated the "Filter" label with the form in some more explicit way, so the controls within the form are differentiated from the top-level search box. I wonder if restructuring this as a [fieldset and legend](https://www.w3.org/TR/2012/NOTE-WCAG20-TECHS-20120103/H71) inside the form might work. What do you think?

Reply:
Yes, this label doesn't contain an input, nor does it point to an input with a `for` attribute, so this label doesn't have any semantic use.

Reply:
Thanks, @crhallberg. Do you think fieldset and legend is a viable alternative?

Reply:
Even with fieldset, you would need `for`. I wouldn't recommend fieldset as an alternative unless we're grouping multiple controls together or we really need a secondary associated label. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset for examples.

Reply:
@crhallberg, we ARE grouping multiple controls together -- we have a search input and a reset button that are both related to filtering the facet list. We need to indicate that these things are all related, and differentiate them from the top-level search box. That's why I suggested a fieldset INSTEAD of a label (since the label can only be connected to one thing). I'm not saying that the fieldset is necessarily the correct/best solution -- that's why I'm asking for input -- but my point is that we basically have a whole unlabeled form that needs to be put into context in an accessible way.

Reply:
Update: I spoke to @crhallberg and I think we've reached consensus that fieldset/legend is appropriate here. I also asked for an opinion from @metageeky, who agreed that fieldset/legend makes sense in this scenario (but can't give this a full stamp of approval without actually seeing it operating in action first). With some help from @crhallberg on the styling, I've pushed up a fieldset/legend implementation here: https://github.com/vufind-org/vufind/pull/2991/commits/d26ad12f4f6d077992752c6955cfe181dbff7558 -- note that this also allowed me to simplify the "hide when JS is disabled" code, since we now only need to hide a single element.